samples/Samples.Mvc5/Controllers/HomeController.cs

```diff
@@ -131,6 +133,35 @@ public ActionResult About()
         /// <returns>The <see cref="ActionResult"/>.</returns>
         public ActionResult ResultsAuthorization() => View();
 
+        public async Task<ActionResult> CallAsync()
+        {
+            var profiler = MiniProfiler.Current;
+            var service = new FooService();
+            using (profiler.Step("action"))
+            {
+                var tasks = new List<Task>();
+                for (int i = 0; i < 10; i++)
+                {
+                    tasks.Add(service.FooAsync());
+                }
+                await Task.WhenAll(tasks);
+                return Content("All good");
+            }
+        }
+
+        class FooService
+        {
+            public async Task FooAsync()
+            {
+                var profiler = MiniProfiler.Current;
```

Comment:
MVC sample is using WebRequestProfilerProvider which relies on HttpContext to carry the miniprofiler instance. Since this is a continuation after ConfigureAwait(false) the synchronization context together with the HttpContext will be removed here. So if I don't use an explicit reference to profiler here the step would not be recorded at all.

DefaultProfilerProvider (and its possible future merge with WebRequestProfilerProvider) would fix this issue by using AsyncLocal to carry the mini profiler instance.

However the execution of the continuation is not synchronized with continuations from other 9 calls to FooAsync that are made from the same CallAsync call (and they all share the same instance of MiniProfiler). As a result Timing.AddChild might be invoked concurrently.

This is the exact problem that i'm trying to prevent.